### PR TITLE
Add support for layer declarations ending with semicolon

### PR DESCRIPTION
**Pull request by samuil-banti-wpenigne**
Semicolon-terminated `@layer` is valid CSS but was parsed as a block, so later rules were swallowed.

`@layer` has two forms ([CSS Cascade 5](https://www.w3.org/TR/css-cascade-5/#layer-empty)):

```
@layer reset;
@layer base, components;
@layer theme { .button { color: blue; } }
```

Generic at-rule parsing always searched for {. For `@layer reset;`, that meant consuming through the next { (e.g. `@property`), so the rest of the stylesheet was treated as nested contents of a fake layer block.

Fix: stop at `{` or `;`. A `;` becomes `AtRuleStatement;` a `{` still becomes `AtRuleBlockList` or `AtRuleSet`. layer stays in `BLOCK_RULES`, so `@layer theme { … }` is unchanged.

### CODE REVIEW

File: tests/CSSList/AtRuleStatementTest.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sabberworm\CSS\Tests\CSSList;
```
Comment (JakeQZ) [inline tests/CSSList/AtRuleStatementTest.php]: I think this should be in the namespace `Sabberworm\CSS\Tests\Functional\Property`, with the file location changed accordingly.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ Please also have a look at our
 
 ### Added
 
+- Add tests for semicolon-terminated `@layer` statements (#1624)
+
```
Comment (JakeQZ) [inline CHANGELOG.md]:
```suggestion
```

This isn't needed as it's developer-specific.  The main entry in the 'Fixed' section is sufficient.

**Review by JakeQZ — CHANGES_REQUESTED** [review]
Wow.  Excellent contribution @samuil-banti-wpenigne.  Hard to pick any holes in that.

The functional tests for the new class should be in a different location (and namespace), and I don't think we need a changelog entry that tests have been added.

Otherwise, looks perfect.  Thank you so much, particularly for taking the time to understand the codebase.

@oliverklee, do you have any other nitpicks?

File: src/CSSList/CSSList.php [hunk 3]
```diff
@@ -210,8 +211,13 @@ private static function parseAtRule(ParserState $parserState): ?CSSListItem
             }
             return new CSSNamespace($url, $prefix, $identifierLineNumber);
         } else {
-            // Unknown other at rule (font-face or such)
-            $arguments = \trim($parserState->consumeUntil('{', false, true));
+            // Unknown other at rule (font-face, @layer, or such)
+            $arguments = \trim($parserState->consumeUntil(['{', ';'], false, false));
```
[the remark this answers is not in the document]
Comment (JakeQZ) [reply src/CSSList/CSSList.php:215]: It's the same with `{`.  Fixing it is beyond the scope of this PR.  #1625 added.

**Review by JakeQZ — APPROVED** [review]
LGTM now.  One of the best PR submissions we've ever had.  Thank you.

@oliverklee, anything else to pick up on?

`phar.io` server is currently down, causing failing checks...

**Comment by oliverklee** [comment]
@JakeQZ I most probably won't have time to look at this today or tomorrow, and I don't want to block this. So feel free to merge in the meantime.

**Comment by JakeQZ** [comment]
> @JakeQZ I most probably won't have time to look at this today or tomorrow, and I don't want to block this. So feel free to merge in the meantime.

OK.  Happy to do so so that people can pick up the `-dev` branch.  The code change looks solid, the new class located with its siblings (which may all be relocated at some point), and the tests good.

The only thing I can think of that might be falling short is whether the tests cover every eventuality and format of at-rules.  But they never will - there's always an edge case that has been missed.  AI was surprisingly smart at picking up the pre-existing lack of string parsing, which has been logged as #1625.  Any other issues found can be resolved as a follow-up.

Thanks again @samuil-banti-wpenigne for your excellent contribution.
